src/transformers/dynamic_module_utils.py

```diff
@@ -200,8 +200,11 @@ def get_class_in_module(class_name: str, module_path: Union[str, os.PathLike]) -
     name = os.path.normpath(module_path).rstrip(".py").replace(os.path.sep, ".")
     module_path = str(Path(HF_MODULES_CACHE) / module_path)
     module_spec = importlib.util.spec_from_file_location(name, location=module_path)
-    module = importlib.util.module_from_spec(module_spec)
+    module = sys.modules.pop(name, None)
+    if module is None:
+        module = importlib.util.module_from_spec(module_spec)
     module_spec.loader.exec_module(module)
+    sys.modules[name] = module
```

Comment:
From the description of [load_module](https://docs.python.org/3/library/importlib.html#importlib.abc.Loader.load_module) (yes, I know it's deprecated):

> If the requested module already exists in [sys.modules](https://docs.python.org/3/library/sys.html#sys.modules), that module should be used and reloaded. Otherwise the loader should create a new module and insert it into [sys.modules](https://docs.python.org/3/library/sys.html#sys.modules) before any loading begins, to prevent recursion from the import. If the loader inserted a module and the load fails, it must be removed by the loader from [sys.modules](https://docs.python.org/3/library/sys.html#sys.modules); modules already in [sys.modules](https://docs.python.org/3/library/sys.html#sys.modules) before the loader began execution should be left alone.

and [this example](https://docs.python.org/3/library/importlib.html#checking-if-a-module-can-be-imported):

```python
    # If you chose to perform the actual import ...
    module = importlib.util.module_from_spec(spec)
    sys.modules[name] = module
    spec.loader.exec_module(module)
    print(f"{name!r} has been imported")
```

here, I would probably do:

```python
    module = sys.modules.get(name, None)
    if module is None:
        module = importlib.util.module_from_spec(module_spec)
        # insert it into sys.modules before any loading begins
        sys.modules[name] = module
    # reload in both cases
    module_spec.loader.exec_module(module)
```

WDYT?